kinto/core/resource/__init__.py

```diff
@@ -342,10 +342,11 @@ def collection_delete(self):
         self._raise_412_if_modified()
 
         filters = self._extract_filters()
+        records, _ = self.model.get_records(filters=filters)
         deleted = self.model.delete_records(filters=filters)
```

Comment:
Is it possible for there to be records returned that are then not removed by the delete (i.e. a race condition with another client)? Or is this all happening in a transaction?